### PR TITLE
Emscripten simd

### PR DESCRIPTION
**Pull request by huningxin**

### CODE REVIEW

**Comment by huningxin** [comment]
@chadaustin , please take a look. Thanks!

File: skinning/skinning_test_simd.cpp [hunk 1]
```diff
@@ -162,7 +162,7 @@ void calculateVerticesAndNormals_SSE_intrinsics(
       _mm_storeh_pi((__m64*)output_vertex, sum2);
 
       __m128 sum3 = _mm_shuffle_ps(sum2, sum2, _MM_SHUFFLE(1, 1, 1, 1));
-      sum3 = _mm_add_ss(sum2, sum3);
+      sum3 = _mm_add_ps(sum2, sum3);
```
Comment (chadaustin) [inline skinning/skinning_test_simd.cpp]: why did you change this? _mm_add_ss on several CPUs is faster, so this isn't quite a fair comparison.

Comment (huningxin) [inline skinning/skinning_test_simd.cpp]: My reason is for emscripten code generation, emscripten will generate one more
SIMD.float32x4.shuffle for _mm_add_ss than for _mm_add_ps. See https://docs.google.com/a/intel.com/spreadsheets/d/1QAGGf2M2IA6l4cvh8eTXdXGEUcPjdmTe_BLKGn5YCB4/edit#gid=0

Comment (huningxin) [inline skinning/skinning_test_simd.cpp]: For example, in my test, with emscripten patch (https://github.com/kripken/emscripten/pull/3120), the number of _mm_add_ss is:
````
/home/nhu/devel/simd.js/v8-m40/out/ia32.release/d8 --simd-object emscripten-O3-simd.js
Skinned vertices per second: 83608000, blah=0.000000
````
The number of _mm_add_ps is:
```
/home/nhu/devel/simd.js/v8-m40/out/ia32.release/d8 --simd-object emscripten-O3-simd.js
Skinned vertices per second: 89698000, blah=0.000000
```

Comment (huningxin) [inline skinning/skinning_test_simd.cpp]: According to https://software.intel.com/sites/landingpage/IntrinsicsGuide/, the `addss` and `addps` has same latency and throughput performance after Nehalem architecture.

**Comment by chadaustin** [comment]
At the instruction level, on modern CPUs that have effectively unlimited execution units, addps and addss have the same performance, though it's conceivable that a compiler could fold multiple addss ops into one addps.  Care to humor me and add an #ifdef EMSCRIPTEN?

**Comment by huningxin** [comment]
Thanks for your explanation. Added `__EMSCRIPTEN__` directive. Please take another look. Thanks.

**Comment by chadaustin** [comment]
Thanks!

**Comment by huningxin** [comment]
@chadaustin , sent an email to chad@chadaustin.me for SIMD optimization discussion. Please have a check. Thanks!
